### PR TITLE
[K8S][WIP] add module kyuubi-cleaner to build daemonset kyuubi-cleaner

### PR DESCRIPTION
**Pull request by zwangsheng**
### _Why are the changes needed?_
add module kyuubi-cleaner to build daemonset kyuubi-cleaner to help clean shuffle data on k8s
Help create daemonSet on work node for k8s to clean shuffle data
use  SHUFFLE_DIRS to config shuffle data dirs
use FILE_TIMEOUT to judging the expiration time
use THRESHOLD to judging if should continue clean

Code Logic：
Determine whether the files under the shuffle directory exceed the specified time, and clear the files that exceed the specified time.
After cleaning up again, the disk capacity still exceeds the threshold, perform another deep cleaning (expiration time is shortened)。
### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by zwangsheng** [comment]
cc @yaooqinn @ulysses-you @pan3793 This is a basic demon, which will be improved in the future, please give suggestions on the code logic

File: kyuubi-cleaner/pom.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>kyuubi</artifactId>
+        <groupId>org.apache.kyuubi</groupId>
+        <version>1.2.0-SNAPSHOT</version>
```
Comment (yaooqinn) [inline kyuubi-cleaner/pom.xml:8]: 1.3

File: kyuubi-cleaner/pom.xml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    <artifactId>kyuubi-cleaner</artifactId>
+    <packaging>jar</packaging>
+    <name>Kyuubi Kubernetes Cleaner</name>
+
+    <build>
+        <outputDirectory>target/scala-${scala.binary.version}/classes</outputDirectory>
+        <testOutputDirectory>target/scala-${scala.binary.version}/test-classes</testOutputDirectory>
+    </build>
+</project>
```
Comment (yaooqinn) [inline kyuubi-cleaner/pom.xml:21]: newline

File: kyuubi-cleaner/pom.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        <artifactId>kyuubi</artifactId>
+        <groupId>org.apache.kyuubi</groupId>
+        <version>1.2.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>kyuubi-cleaner</artifactId>
+    <packaging>jar</packaging>
+    <name>Kyuubi Kubernetes Cleaner</name>
```
Comment (yaooqinn) [inline kyuubi-cleaner/pom.xml:15]: Kubernetes Cleaner？Clean what？

File: kyuubi-cleaner/src/main/resources/META-INF/MANIFEST.MF [hunk 4]
```diff
@@ -0,0 +1,3 @@
+Manifest-Version: 1.0
```
Comment (yaooqinn) [inline kyuubi-cleaner/src/main/resources/META-INF/MANIFEST.MF:1]: necessary？

File: kyuubi-cleaner/src/main/scala/kubernetes/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,5 @@
+FROM openjdk:8-jre-slim
+COPY kyuubi-cleaner-1.2.0-SNAPSHOT.jar /opt/kyuubi-cleaner-1.2.0-SNAPSHOT.jar
+COPY entrypoint.sh /opt/entrypoint.sh
+RUN mkdir /data
+ENTRYPOINT ["/opt/entrypoint.sh"]
```
Comment (yaooqinn) [inline kyuubi-cleaner/src/main/scala/kubernetes/Dockerfile:5]: newline

File: kyuubi-cleaner/src/main/scala/kubernetes/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,5 @@
+FROM openjdk:8-jre-slim
+COPY kyuubi-cleaner-1.2.0-SNAPSHOT.jar /opt/kyuubi-cleaner-1.2.0-SNAPSHOT.jar
```
Comment (yaooqinn) [inline kyuubi-cleaner/src/main/scala/kubernetes/Dockerfile:2]: hardcode？

File: kyuubi-cleaner/src/main/scala/kubernetes/cleaner.yaml [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  selector:
+    matchLabels:
+      name: colocation-service
+  template:
+    metadata:
+      labels:
+        name: colocation-service
+    spec:
+      containers:
+        - image: harbor-beta.cm.service.163.org/spark/kyuubi-cleaner:0.0.21
```
Comment (yaooqinn) [inline kyuubi-cleaner/src/main/scala/kubernetes/cleaner.yaml:16]: what is this for？

File: kyuubi-cleaner/src/main/scala/kubernetes/entrypoint.sh [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+java -cp /opt/kyuubi-cleaner-1.2.0-SNAPSHOT.jar \
+ org.apache.kyuubi.Main \
```
Comment (yaooqinn) [inline kyuubi-cleaner/src/main/scala/kubernetes/entrypoint.sh:20]: confusing?

**Pull request closed without merging**